crates/cornucopia/src/container.rs

```diff
@@ -2,58 +2,93 @@ use std::process::{Command, Stdio};
+    #[clap(long = "container-name", default_value = DEFAULT_CONTAINER_NAME)]
+    pub container_name: String,
+    #[clap(long = "docker-image", default_value = DEFAULT_DOCKER_IMAGE)]
+    pub docker_image: String,
+}
+
+impl Default for ContainerOpts {
+    fn default() -> Self {
+        Self {
+            podman: false,
+            port: DEFAULT_PORT,
+            container_name: DEFAULT_CONTAINER_NAME.into(),
+            docker_image: DEFAULT_DOCKER_IMAGE.into(),
+            // startup_delay: STARTUP_DELAY,
+        }
+    }
+}
+
 /// Starts Cornucopia's database container and wait until it reports healthy.
-pub fn setup(podman: bool) -> Result<(), Error> {
-    spawn_container(podman)?;
-    healthcheck(podman, 120, 50)?;
+pub fn setup(opts: &ContainerOpts) -> Result<(), Error> {
+    spawn_container(opts)?;
+    healthcheck(opts, 120, 50)?;
     Ok(())
 }
 
 /// Stop and remove a container and its volume.
-pub fn cleanup(podman: bool) -> Result<(), Error> {
-    stop_container(podman)?;
-    remove_container(podman)?;
+pub fn cleanup(opts: &ContainerOpts) -> Result<(), Error> {
+    stop_container(opts)?;
+    remove_container(opts)?;
     Ok(())
 }
 
 /// Starts Cornucopia's database container.
-fn spawn_container(podman: bool) -> Result<(), Error> {
-    cmd(
-        podman,
-        &[
-            "run",
-            "-d",
-            "--name",
-            "cornucopia_postgres",
-            "-p",
-            "5435:5432",
-            "-e",
-            "POSTGRES_PASSWORD=postgres",
-            "docker.io/library/postgres:latest",
-        ],
-        "spawn container",
-    )
+fn spawn_container(opts: &ContainerOpts) -> Result<(), Error> {
+    let args = [
+        "run",
+        "-d",
+        "--name",
+        &opts.container_name,
+        "-p",
+        &format!("{}:5432", opts.port),
+        "-e",
+        "POSTGRES_PASSWORD=postgres",
+        &opts.docker_image,
+    ];
+    cmd(opts, &args, "spawn container")
 }
 
 /// Checks if Cornucopia's container reports healthy
-fn is_postgres_healthy(podman: bool) -> Result<bool, Error> {
+fn is_postgres_healthy(opts: &ContainerOpts) -> Result<bool, Error> {
     Ok(cmd(
-        podman,
-        &["exec", "cornucopia_postgres", "pg_isready"],
+        opts,
+        &["exec", &opts.container_name, "pg_isready"],
         "check container health",
     )
     .is_ok())
 }
 
 /// This function controls how the healthcheck retries are handled.
-fn healthcheck(podman: bool, max_retries: u64, ms_per_retry: u64) -> Result<(), Error> {
+fn healthcheck(opts: &ContainerOpts, max_retries: u64, ms_per_retry: u64) -> Result<(), Error> {
     let slow_threshold = 10 + max_retries / 10;
     let mut nb_retries = 0;
-    while !is_postgres_healthy(podman)? {
+    let mut consecutive_successes = 0;
+    while consecutive_successes < 5 {
+        if is_postgres_healthy(opts)? {
+            consecutive_successes += 1;
+        } else {
+            consecutive_successes = 0;
+        }
```

Comment:
Checking the container logs to see if the "PostgreSQL init process complete; ready for start up." line has been printed is probably more robust than relying on an arbitrary amount of successes.